go/arrow/array/map.go

```diff
@@ -126,26 +126,25 @@ type MapBuilder struct {
 // building using keys in sorted order for each value. The KeysSorted value will just be
 // used when creating the DataType for the map.
 //
-// Example
+// # Example
```

Comment:
Go documentation doesn't need the `#` here. A single line without a period becomes a header like so: https://pkg.go.dev/github.com/apache/arrow/go/v10@v10.0.1/arrow/array#hdr-Example